File: 04-janelas-graficas/01-janela-basica-manual/JanelaBasica.java
package visao;

import java.awt.*; 
import java.awt.event.*; 
import javax.swing.*;

public class JanelaBasica {

    public static void criar() {
        // criar a janela e configurar tamanho e ação de fechamento
        JFrame j = new JFrame("Janela básica");
        j.setPreferredSize(new Dimension(250, 130));
        j.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        // configurar layout da janela
        j.getContentPane().setLayout(new FlowLayout());
        // criar um texto (rótulo)
        JLabel msg = new JLabel("Digite o seu nome: ");
        // adicionar a mensagem na janela
        j.getContentPane().add(msg);
        // criar uma caixa de texto e adicionar na janela
        JTextField nome = new JTextField("", 10);
        j.getContentPane().add(nome);
        // criar um botão e adicionar na janela
        JButton but = new JButton("OK");
        j.getContentPane().add(but);
        // criar outro rótulo e adicionar na janela
        JLabel msg2 = new JLabel("");
        j.getContentPane().add(msg2);
        // adicionar um tratador de evento ao botão
        but.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent arg0) {
                // colocar no segundo rótulo o título da caixa de texto
                msg2.setText("Seu nome é: "+nome.getText());
            }
        });
        // operações finais sobre a janela
        j.setLocationRelativeTo(null); // centralizar
        j.pack(); // montar 
        j.setVisible(true); // tornar visível
    }
    public static void main(String[] args) {
        JanelaBasica.criar();
    }
}

File: 04-janelas-graficas/01-janela-basica-manual/janela-basica.py
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

class JanelaBasica(Gtk.Window):

    def __init__(self):
        # executar o método construtor da janela
        Gtk.Window.__init__(self, title="Janela básica")
        # criar um layout vertical, 6 pixels entre cada componente
        box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=6)
        box.set_property("margin", 8) # adicionar uma borda na janela
        self.add(box) # inserir o layout na janela
        self.msg = Gtk.Label("Digite o seu nome: ") # criar um rótulo na janela
        box.pack_start(self.msg, True, True, 0) # adicionar o rótulo ao layout
        # criar um layout horizontal para por caixa de entrada ao lado do botão
        hbox = Gtk.Box() # padrão: horizontal
        # criar uma caixa de texto e adicioná-la ao layout horizontal
        self.nome = Gtk.Entry()
        hbox.pack_start(self.nome, True, True, 0)
        self.button = Gtk.Button(label="Ok") # criar botão "Ok"
        # associar evento de clique ao botão
        self.button.connect("clicked", self.acao_clique)
        # adicionar botão ao layout horizontal
        hbox.pack_start(self.button, True, True, 0)
        # adicionar layout horizontal ao layout vertical
        box.pack_start(hbox, True, True, 0)
        # criar rótulo na janela para mensagem pós-clique
        self.msg2 = Gtk.Label()
        self.msg2.set_width_chars(20)
        # adicionar no layout vertical
        box.pack_start(self.msg2, True, True, 0)

    #  evento de clique no botão
    def acao_clique(self, widget):
        # colocar no segundo rótulo o conteúdo da caixa de texto
        self.msg2.set_label("Seu nome é: "+self.nome.get_text())

win = JanelaBasica() # instanciar janela
win.connect("destroy", Gtk.main_quit) # encerrar o programa ao fechar a janela
win.set_position(Gtk.WindowPosition.CENTER) # centralizar janela
win.show_all() # exibir janela
Gtk.main() # início do gtk (loop contínuo)